app/controllers/categories_controller.rb

```diff
@@ -62,4 +63,10 @@ def set_category
     def category_params
       params.require(:category).permit(:namecategory)
     end
+
+    def authenticate_user
+      if current_user.nil? || current_user.email != "procuponss@gmail.com"
```

Comment:
I think that instead of the email, you should put an environment variable.

```suggestion
      if current_user.nil? || current_user.email != EMAIL_ADMIN
```